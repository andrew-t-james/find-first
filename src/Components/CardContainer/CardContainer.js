import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';

import Card from '../Card/Card';

export const CardContainer = ({ jobs }) => {
  const renderJobs = jobs.map(job =>
    <Link to={`/job/${job.id}`} key={job.id} >
      <Card {...job} key={job.id}/>
    </Link>
  );

  return (
    <section className="cards-section">
      {jobs.length ? renderJobs : null}
    </section>
  );
};

export default CardContainer;

CardContainer.propTypes = {
  jobs: PropTypes.arrayOf(PropTypes.object)
};