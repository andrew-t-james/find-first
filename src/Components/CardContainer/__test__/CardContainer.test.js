import React from 'react';
import { shallow } from 'enzyme';
import { CardContainer, mapStateToProps } from '../CardContainer';
describe('<CardContainer />', () => {
  let wrapper;
  const mockJobs = [
    {
      id: 1,
      title: 'A dope dev job',
      location: 'remote'
    },
    {
      id: 2,
      title: 'Another dope dev job',
      location: 'remote'
    }
  ];

  beforeEach(() => wrapper = shallow(<CardContainer jobs={mockJobs} />));

  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
