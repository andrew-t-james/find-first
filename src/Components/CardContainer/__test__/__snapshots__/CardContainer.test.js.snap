// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CardContainer /> should match snapshot 1`] = `
<section
  className="cards-section"
>
  <Card
    id={1}
    key="A dope dev job"
    location="remote"
    title="A dope dev job"
  />
  <Card
    id={2}
    key="Another dope dev job"
    location="remote"
    title="Another dope dev job"
  />
</section>
`;
